python/pyspark/sql/tests.py

```diff
@@ -545,6 +545,15 @@ def test_udf_in_filter_on_top_of_join(self):
         right = self.spark.createDataFrame([Row(b=1)])
         f = udf(lambda a, b: a == b, BooleanType())
         df = left.crossJoin(right).filter(f("a", "b"))
+
+    def test_udf_in_join_condition(self):
+        # regression test for SPARK-25314
+        from pyspark.sql.functions import udf
+        left = self.spark.createDataFrame([Row(a=1)])
+        right = self.spark.createDataFrame([Row(b=1)])
+        f = udf(lambda a, b: a == b, BooleanType())
+        df = left.crossJoin(right).filter(f("a", "b"))
```

Comment:
BTW, why do we explicitly test cross join here?

Reply:
ditto, the correct test is `df = left.join(right, f("a", "b"))`.